Sources/Clappr/Classes/Base/Core.swift

```diff
@@ -219,14 +223,46 @@ open class Core: UIObject, UIGestureRecognizerDelegate {
 
         trigger(.didDestroy)
     }
+}
 
-    private func safeDestroy(_ plugin: Plugin) {
+fileprivate extension Plugin {
+    func logRenderCrash(for error: Error) {
+        Logger.logError("\(pluginName) crashed during render (\(error.localizedDescription))", scope: "Core")
+    }
+
+    func logDestroyCrash(for error: Error) {
+        Logger.logError("\(pluginName) crashed during destroy (\(error.localizedDescription))", scope: "Core")
+    }
+
+    var shouldFitParentView: Bool {
+        return (self as? OverlayPlugin)?.isModal == true
+    }
+
+    #if os(iOS)
+    var isNotMediaControlElement: Bool {
+        return !(self is MediaControl.Element)
+    }
+    #endif
+
+    func safeDestroy() {
         do {
             try ObjC.catchException {
-                plugin.destroy()
+                self.destroy()
+            }
+        } catch {
+            logDestroyCrash(for: error)
+        }
+    }
+
+    func safeRender() {
```

Comment:
Once you have typecasted `plugin` before using it, wouldn't it make more sense to add this method to a `UICorePlugin` extension?